File: migrate-csv.mjs
import { sql } from './src/lib/db.js';
import fs from 'fs';
import path from 'path';

async function migrate() {
    try {
        const csvPath = path.join(process.cwd(), 'kghs-89f golf data.csv');
        const data = fs.readFileSync(csvPath, 'utf8');
        const rows = data.split('\n').map(row => row.split(','));

        // Row 0: Name, 강정석, 김기홍, ...
        const names = rows[0].slice(3).map(n => n.trim()).filter(n => n.length > 0);
        // Row 2: count, Date, CC/HD, 94.3, ...
        const handicaps = rows[2].slice(3).map(h => h.trim());

        console.log(`${names.length}명의 회원 데이터를 처리 중...`);

        for (let i = 0; i < names.length; i++) {
            const name = names[i];
            const hdStr = handicaps[i] || '0';
            const hd = parseFloat(hdStr) || 0;

            await sql`
            INSERT INTO members (name, handicap, role)
            VALUES (${name}, ${hd}, 'member')
            ON CONFLICT (id) DO NOTHING
        `;
            console.log(`이동 완료: ${name} (HD: ${hd})`);
        }

        console.log('마이그레이션 성공!');
    } catch (error) {
        console.error('마이그레이션 실패:', error);
    }
}

migrate();

File: migrate_csv.py
import csv
import os
import psycopg2
from dotenv import load_dotenv

def migrate_members():
    # Load environment variables from .env.local
    load_dotenv('.env.local')
    database_url = os.getenv('DATABASE_URL')
    
    if not database_url:
        print("DATABASE_URL not found in .env.local")
        return

    try:
        conn = psycopg2.connect(database_url)
        cur = conn.cursor()
        
        with open('kghs-89f golf data.csv', mode='r', encoding='utf-8') as f:
            reader = list(csv.reader(f))
            
            # Row 0: Empty, Empty, Name, 강정석, ...
            # Row 2: count, Date, CC/HD, 94.3, ... (Handicaps)
            names = reader[0][3:]
            handicaps = reader[2][3:]
            
            inserted_count = 0
            for i in range(len(names)):
                name = names[i].strip()
                if not name: continue
                
                try:
                    # Parse handicap
                    hd_str = handicaps[i].strip() if i < len(handicaps) else '0'
                    hd = float(hd_str) if hd_str else 0.0
                except ValueError:
                    hd = 0.0
                    
                cur.execute(
                    "INSERT INTO members (name, handicap, role) VALUES (%s, %s, %s) ON CONFLICT DO NOTHING",
                    (name, hd, 'member')
                )
                inserted_count += 1
                
        conn.commit()
        print(f"Successfully migrated {inserted_count} members.")
        
    except Exception as e:
        print(f"Error: {e}")
    finally:
        if conn:
            cur.close()
            conn.close()

if __name__ == "__main__":
    migrate_members()
